### PR TITLE
chore: lerna publish workflow tweaks

### PR DESCRIPTION
**Pull request by joeldodge79**
easier to setup/use node/npm as actions/setup-node@v2 already handles
the npm auth step using the NODE_AUTH_TOKEN env var. Also, the wombat
dressing room proxy can be problematic when using yarn.

npm install:
--no-verify-access: verification through the wombat proxy was not happy
--no-save, --no-package-lock: lerna does not like untracked files
--legacy-peer-deps: npm7 cli bombs out otherwise (react library gremlins)

package.json:

- adding a top level package to release-please so we can key off a single
  tag for publishing the typescript packages (e.g. sdk-codegen-all-v1.1.0)
- removed unneeded `yarn` command from build scripts

prepublishOnly support: setup the testenv

lerna.json: there are still cases where we'll do a
```
lerna publish --canary
```

(potentially with the `--no-git-tag-version` flag?) so we want each
package version to just bump independently in that case.

### CODE REVIEW

File: .github/workflows/lerna-publish.yml
Comment (jkaster) [reply .github/workflows/lerna-publish.yml:26]: I think @bryans99 understands this packing stuff pretty well also. WRT `prepare` vs. `build` I just went with what we had in other repos.

**Review by jkaster — DISMISSED** [review]
I don't think there's any further work to do on this PR?

File: .github/workflows/lerna-publish.yml
Comment (lukelooker) [reply .github/workflows/lerna-publish.yml:26]: Sorry, thought I sent a comment but apparently not. `prepare` is a "postinstall" hook in Yarn 1 but it's gone away in Yarn 2 so I didn't want to build an explicit expectation of it.

Also, in our repo `build` takes a while so I wanted to only run it when needed, not on every `yarn` action.

Comment (lukelooker) [reply .github/workflows/lerna-publish.yml:26]: Our equivalent to this file is: https://github.com/looker-open-source/components/blob/main/.github/workflows/release.yml

It does:

1. yarn installl
2. get lerna version
3. check if github "release" exists (STOP if already exists)
5. build packages
4. create Github "release"
6. publish NPM packages
7. build web site
8. publish web site

Comment (lukelooker) [reply .github/workflows/lerna-publish.yml:26]: Ok - found useful links:

Yarn 1: https://classic.yarnpkg.com/en/docs/package-json#scripts-
Yarn 2: https://yarnpkg.com/advanced/lifecycle-scripts

Comment (joeldodge79) [reply .github/workflows/lerna-publish.yml]: ah, I didn't realize automatically calling `prepare` had been removed in yarn2. I'll stick with `build`

**Comment by joeldodge79** [comment]
> I don't think there's any further work to do on this PR?

sorry, still tweaking it @jkaster - I'll ping you when it's ready

File: .github/workflows/lerna-publish.yml
Comment (jkaster) [inline .github/workflows/lerna-publish.yml:63]: FWIW if we've got ssl always working I don't think we need to document this. But it certainly doesn't hurt anything.

File: package.json
Comment (jkaster) [inline package.json:37]: Looks good to me.

**Review by jkaster — APPROVED** [review]
Major step forward!

File: package.json
Comment (joeldodge79) [reply package.json:37]: FYI I copied the `"clean"` script into the `"build"` script because I'm trying to avoid calling `yarn` from inside any script that needs to run as part of the github action publish flow since it's using npm (due to wombat dressing room compatibility issues).

One way to DRY that out would be to create a `clean.sh` executable script and then call `./clean.sh` in both `"build"` and `"clean"`

File: .github/workflows/lerna-publish.yml
Comment (joeldodge79) [reply .github/workflows/lerna-publish.yml:63]: shoot, all `methods.spec.ts` tests failing like this:
```
● LookerNodeSDK › sudo › login/logout

    POST /api/3.1/login <!DOCTYPE html>
    <html>
    <head>
        <title>Looker Unavailable (502)</title>
        <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,600' rel='stylesheet' type='text/css'>
        <link rel="icon" type="image/x-icon"  href="data:image/png;base64,AAABAAIAEBAAAAEAIABoBAAAJgAAACAgAAABACAAqBAAAI4EAAAoAAAAEAAAACAAAAABACAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAO+HQALzhEFO84RBvvOEQfLzhEHy84RBvPOEQUwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAO6HQALzhEGG9IVB//OEQu7zhEKu84RBrvOFQfDzhEH/84RBggAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADzhEFM84VB//SFQbjzhEESAAAAAAAAAADzhUES84VBuvOEQf/zhEFGAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA84RBvPOEQvDzhEESAAAAAAAAAAAAAAAAAAAAAPOFQRTzhEHy84RBtgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPOEQe7zhEGyAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA84RBtvOEQuoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADzhEHu84RBtAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPOEQbb0hEHoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA84RBtvOEQfLzhEEWAAAAAAAAAAAAAAAAAAAAAPOEQRjzhEH084RBsgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPOFQUTzhUH/84RBwvOEQRYAAAAAAAAAAPOEQRrzhEHE84VB//OEQUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA84VBePOEQf/zhEE484RBsPOEQbzzhEH284RB//OEQnQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADVC6SCgaIta84RBPPOEQuTzhELk84RBrvOEQT7ugEACAAAAAAAAAAAAAA...
        <style type="text/css">
          body ***
            background-color: #2e343f;
            color: white;
            height: auto;
            font-family: Open Sans, Helvetica, Arial, sans-serif;
          ***
          .message ***
            width: 100%;
            max-width: 760px;
            margin: 0 auto;
            margin-top: 135px;
            text-align: center;
          ***
          h2, h3 ***
            font-weight: normal;
          ***
          a ***
            color: white;
          ***
        </style>
    </head>
    <body>

      <div class="message">

        <img width="210" height="84" alt="Looker" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAaQAAACoCAYAAACv+Zr6AAAACXBIWXMAAAsTAAALEwEAmpwYAAA4KmlUWHRYTUw6Y29tLmFkb2JlLnhtcAAAAAAAPD94cGFja2V0IGJlZ2luPSLvu78iIGlkPSJXNU0wTXBDZWhpSHpyZVN6TlRjemtjOWQiPz4KPHg6eG1wbWV0YSB4bWxuczp4PSJhZG9iZTpuczptZXRhLyIgeDp4bXB0az0iQWRvYmUgWE1QIENvcmUgNS42LWMxMzIgNzkuMTU5Mjg0LCAyMDE2LzA0LzE5LTEzOjEzOjQwICAgICAgICAiPgogICA8cmRmOlJERiB4bWxuczpyZGY9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkvMDIvMjItcmRmLXN5bnRheC1ucyMiPgogICAgICA8cmRmOkRlc2NyaXB0aW9uIHJkZjphYm91dD0iIgogICAgICAgICAgICB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iCiAgICAgICAgICAgIHhtbG5zOmRjPSJodHRwOi8vcHVybC5vcmcvZGMvZWxlbWVudHMvMS4xLyIKICAgICAgICAgICAgeG1sbnM6cGhvdG9zaG9wPSJodHRwOi8vbnMuYWRvYmUuY29tL3Bob3Rvc2hvcC8xLjAvIgogICAgICAgICAgICB4bWxuczp4bXBNTT0iaHR0cDovL25zLmFkb2JlLmNvbS94YXAvMS4wL21tLyIKICAgICAgICAgICAgeG1sbnM6c3RFdnQ9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9zVHlwZS9SZXNvdXJjZUV2ZW50IyIKICAgICAgICAgICAgeG1sbnM6dGlmZj0iaHR0cDovL25zLmFkb2JlLmNvbS90aWZmLzE...

        <h1>Looker is unavailable.</h1>

        <h2>If you typed in a URL, double-check the spelling.</h2>
          <h2>This may also be due to a temporary condition such as an outage, <a href="https://docs.looker.com/relnotes/hosted-maintenance-hours">scheduled maintenance</a> or upgrade.</h2>
        <br>
        <h3>
            If this message persists or you have any concerns, <br> contact us
            from <a href="https://help.looker.com">help.looker.com</a> and we'll respond promptly.
        </h3>

      </div>

    </body>
    </html>

      200 |
      201 |     if ('message' in _error) ***
    > 202 |       return new Error(response.error.message.toString());
          |              ^
      203 |     ***
      204 |   ***
      205 |

      at sdkError (packages/sdk-rtl/lib/transport.js:202:14)
      at packages/sdk-rtl/lib/transport.js:225:13
      at asyncGeneratorStep (packages/sdk-rtl/lib/transport.js:19:103)
      at _next (packages/sdk-rtl/lib/transport.js:21:194)
```

Comment (joeldodge79) [reply .github/workflows/lerna-publish.yml:63]: this is the env:

```
Run $(npm bin)/lerna publish from-package --yes --no-verify-access
  $(npm bin)/lerna publish from-package --yes --no-verify-access
  shell: /usr/bin/bash -e ***0***
  env:
    LOOKERSDK_CLIENT_ID: ***
    LOOKERSDK_CLIENT_SECRET: ***
    NPM_CONFIG_USERCONFIG: /home/runner/work/_temp/.npmrc
    NODE_AUTH_TOKEN: ***
    CLOUDSDK_METRICS_ENVIRONMENT: github-actions-setup-gcloud
    GCLOUD_PROJECT: ***
    GOOGLE_APPLICATION_CREDENTIALS: /home/runner/work/sdk-codegen/sdk-codegen/55f8ca5b-474f-4367-bdc0-f2c92df49517
    pythonLocation: /opt/hostedtoolcache/Python/3.9.2/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.9.2/x64/lib
```

Comment (joeldodge79) [reply .github/workflows/lerna-publish.yml:63]: ah, missing these from the tssdk-ci.yml workflow:
```
  LOOKERSDK_BASE_URL: https://localhost:20000
  LOOKERSDK_VERIFY_SSL: false
```

but not sure how I'm even getting a response at all? how does the github runner possibly have access to an instance at `https://self-signed.looker.com:19999` ?

Comment (joeldodge79) [reply .github/workflows/lerna-publish.yml:63]: whoohoo, that fixed it!

File: .github/workflows/lerna-publish.yml
Comment (joeldodge79) [inline .github/workflows/lerna-publish.yml:7]: when we feel confident in this publish flow we'll add a `on: release` trigger so it goes automatically once the release-pr is merged and the release-please github action has created tags

**Comment by joeldodge79** [comment]
@jkaster this is ready to go now

**Comment by joeldodge79** [comment]
@jkaster sorry, need one more approval on this one

**Review by jkaster — APPROVED** [review]
Nice work

**Pull request merged**
